herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLog.java

```diff
@@ -246,8 +249,8 @@ public void close() throws LogNotAvailableException {
             try {
                 LOGGER.log(Level.INFO, "{0} closing ledger {1}, with LastAddConfirmed={2}, LastAddPushed={3} length={4}, errorOccurred:{5}",
                         new Object[]{tableSpaceDescription(), out.getId(), out.getLastAddConfirmed(), out.getLastAddPushed(), out.getLength(), errorOccurredDuringWrite});
-                out.close();
-            } catch (InterruptedException | org.apache.bookkeeper.client.api.BKException err) {
+                out.closeAsync().get(60, TimeUnit.SECONDS);
+            } catch (InterruptedException | ExecutionException | TimeoutException err) {
```

Comment:
usually it i s better to unwrap ExecutionException.getCause() in order to get rid of the intermediate ExecutionException.

We should also make the timeout configurable